agent/consul/state/catalog.go

```diff
@@ -2888,17 +2897,17 @@ func (s *Store) ServiceTopology(
 		maxIdx = idx
 	}
 
+	// The given service is a destination relative to its downstreams
+	_, intentions, err = s.IntentionMatchOne(ws, matchEntry, structs.IntentionMatchDestination)
```

Comment:
I think to avoid an accident the `intentions` variable should be renamed here so the source and destination variants are kept separate to avoid an accident.